oauthproxy_test.go

```diff
@@ -2240,6 +2241,237 @@ func TestTrustedIPs(t *testing.T) {
 	}
 }
 
+func Test_buildRoutesAllowlist(t *testing.T) {
+	testCases := []struct {
+		name            string
+		skipAuthRegex   []string
+		skipAuthRoutes  []string
+		expectedMethods []string
+		expectedRegexes []string
```

Comment:
Could use a dummy struct for this? 

```
type testAllowedRoute struct {
  method string
  regexString string
}

...

assert.Equal(t, route.method, expectedRoute[i].method)
assert.Equal(t, route.regexp.String(), expectedRoute[i].regexString)
```

The benefit of this being that the method and regex are paired, which I would say makes the test easier to read, WDYT?